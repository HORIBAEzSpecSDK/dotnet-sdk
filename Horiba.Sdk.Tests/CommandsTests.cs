namespace Horiba.Sdk.Tests;

public class CommandsTests
{
    [Fact]
    public async Task GivenIclInfoCommand_WhenSerializing_ThenCommandIdAndNameAreCorrect()
    {
        // Arrange
        var iclInfoCommand = new IclInfoCommand();

        // Act
        var jsonCommand = iclInfoCommand.ToJson();

        // Assert
        jsonCommand.MatchSnapshot(options => options.IgnoreField("id"));
    }

    [Fact]
    public async Task GivenDeviceInfoCommand_WhenSendingUsingCommunicator_ThenReceivesExpectedResponse()
    {
        // Arrange
        using var manager = new DeviceManager();
        await manager.StartAsync(true, false);

        // Act
        var response = await manager.Communicator.SendWithResponseAsync(new IclInfoCommand());

        // Assert
        response.MatchSnapshot(options => options.IgnoreField("id")
            .IgnoreField("**.nodeBuilt")
            .IgnoreField("**.nodeVersion"));
    }

    [Fact]
    public async Task GivenCommand_WhenSerializing_ThenHasCorrectParameter()
    {
        // Arrange
        var command = new IclBinaryModeAllCommand();

        // Act
        var json = command.ToJson();

        // Assert
        json.MatchSnapshot(options => options.IgnoreField("id"));
    }
}