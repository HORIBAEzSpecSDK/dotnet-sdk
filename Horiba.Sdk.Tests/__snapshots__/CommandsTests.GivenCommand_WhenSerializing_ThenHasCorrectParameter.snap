{"id":1,"command":"icl_binMode","parameters":{"mode":"all"}}
